.azure/templates/jobs/default-build.yml

```diff
@@ -29,55 +31,82 @@
+#   codeSign: boolean
+#       This build definition is enabled for code signing. (Only applies to Windows)
 
 #
 # See https://docs.microsoft.com/en-us/vsts/pipelines/yaml-schema for details
 #
 
 parameters:
   agentOs: 'Windows'
-  # phaseName: ''
-  queueName: ''
+  poolName: ''
   buildArgs: ''
   configuration: 'Release'
   demands: []
   beforeBuild: []
   afterBuild: []
+  codeSign: false
   variables: {}
   dependsOn: ''
+  # jobName: '' - use agentOs by default.
+  # jobDisplayName: '' - use agentOs by default.
   # matrix: {} - don't define an empty object default because there is no way in template expression yet to check "if isEmpty(parameters.matrix)"
   artifacts:
     publish: true
-    path: 'artifacts/'  # TODO: this is going to change when we converge with dotnet/arcade tooling
+    path: 'artifacts/'
 
-phases:
-- phase: ${{ coalesce(parameters.phaseName, parameters.agentOs) }}
+jobs:
+- job: ${{ coalesce(parameters.jobName, parameters.agentOs) }}
+  displayName: ${{ coalesce(parameters.jobDisplayName, parameters.agentOs) }}
   dependsOn: ${{ parameters.dependsOn }}
-  queue:
-    # If a matrix of builds has been configured, run the matrix in parallel
+  workspace:
+    clean: all
+  strategy:
     ${{ if ne(parameters.matrix, '') }}:
-      parallel: 4 # Pick 4 as the default because we usually don't have a matrix of more than 4 configs, and there is no way to say parallel: all
+      maxParallel: 8
       matrix: ${{ parameters.matrix }}
-    # Map friendly OS names to the right queue
-    ${{ if ne(parameters.queueName, '') }}:
-      name: ${{ parameters.queueName }}
-    ${{ if and(eq(parameters.queueName, ''), eq(parameters.agentOs, 'macOS')) }}:
-      name: Hosted macOS Preview
-    ${{ if and(eq(parameters.queueName, ''), eq(parameters.agentOs, 'Linux')) }}:
-      name: Hosted Linux Preview
-    ${{ if and(eq(parameters.queueName, ''), eq(parameters.agentOs, 'Windows')) }}:
-      name: Hosted VS2017
-    demands: ${{ parameters.demands }}
+  # Map friendly OS names to the right queue
+  pool:
+    ${{ if ne(parameters.poolName, '') }}:
+      name: ${{ parameters.poolName }}
+    ${{ if and(eq(parameters.poolName, ''), eq(parameters.agentOs, 'macOS')) }}:
+      name: Hosted macOS
+      vmImage: macOS-10.13
+    ${{ if and(eq(parameters.poolName, ''), eq(parameters.agentOs, 'Linux')) }}:
+      name: Hosted Ubuntu 1604
+      vmImage: ubuntu-16.04
+    ${{ if and(eq(parameters.poolName, ''), eq(parameters.agentOs, 'Windows')) }}:
+      ${{ if ne(parameters.codeSign, 'true') }}:
+        name: Hosted VS2017
+        vmImage: vs2017-win2016
+      ${{ if eq(parameters.codeSign, 'true') }}:
+        name: DotNetCore-Windows
   variables:
     AgentOsName: ${{ parameters.agentOs }}
-    ASPNETCORE_TEST_LOG_DIR: ${{ format('{0}/logs/testlogs', parameters.artifacts.path) }}
+    ASPNETCORE_TEST_LOG_DIR: $(Build.ArtifactStagingDirectory)/testlogs
```

Comment:
Isn't this going to cause the long path issues again?

Reply:
I'm not sure I understood why it failed before, but thought we'd give this a try. The file path is at least 16 characters shorter (removes "artifacts/logs/" from the path)

Reply:
It's not the location we put the logs that was the problem, it's the way VSTS zips them up that caused issues. They create a new subfolder and try to copy the logs into it which can increase the path past the limit.

Reply:
Thanks for clarifying. I'm going to remove ASPNETCORE_TEST_LOG_DIR while we investigate workarounds.

Btw, for faster iteration on this, you can set ASPNETCORE_TEST_LOG_DIR and add the upload tasks in your repo's .yml files instead of needing to update the template. When we find a solution that works, we can add that to the template.

Reply:
👍 I'll keep that in mind while we discuss with the VSTS folks